compiler/testData/codegen/box/reflection/callBy/inlineClasses/nullableObject/defaultArguments/jvmStaticFunctionsOnCompanionObject.kt

```diff
@@ -0,0 +1,219 @@
+// TARGET_BACKEND: JVM
+// JVM_TARGET: 1.8
+// IGNORE_BACKEND: JVM_IR
```

Comment:
I find it a bit confusing that we're mixing `IGNORE_BACKEND` with `assertFailsWith`. What if the test continues to be muted even after we fix the underlying issues, only because some `assertFailsWith` in it fails and we forget to update it?

Can we maybe remove all `assertFailsWith` from tests which are ignored by `// IGNORE_BACKEND: JVM_IR`?

Reply:
Does that mean that even the fayling test is unnecessary and should be commented out?

I understand that while some cases have been fixed since 2.0, but there are still cases of failure.
Therefore, I feel that it is better to have a faling test.

I am a little confused.

Reply:
No, failing tests are still necessary. I'm just saying that either they need to be muted completely with `IGNORE_BACKEND`, or partially with `assertFailsWith`. Using both is confusing and risky. So for each new test, let's do either `IGNORE_BACKEND`, or `assertFailsWith`, but not both.

Reply:
OK.

There are both cases of failure in the `JVM_IR` backend and cases of failure in all backends.
Therefore, it should not be possible to unify one or the other.

Personally, I would like to unify it with the failing test, but in this case the results would be divided according to the backend.
Or am I misunderstanding something?

Reply:
If I change the specification of an existing test to `TARGET_BACKEND: JVM_IR`, for example, the following cases will fail (and several others).
https://github.com/k163377/kotlin/blob/43d42f8aa6f62e6b33ed6c2381997b2f259d0c9b/compiler/testData/codegen/box/reflection/callBy/inlineClasses/nullableObject/defaultArguments/jvmStaticFunctionsOnCompanionObject.kt#L105

I assumed that the backend that the test fails on is old, does this mean that a destructive change in the new backend has been detected?
Also, is it correct to change all `IGNORE_BACKEND: JVM_IR` to `TARGET_BACKEND: JVM_IR`, and to treat all failures as failing tests?

I have pushed the above fix for the time being.
43fc242

Reply:
Yes, if some code worked on the old JVM backend but fails on the JVM_IR backend, it most likely means it's a bug in the JVM_IR backend.

It is fine to use `TARGET_BACKEND: JVM_IR` for all new tests, and wrap all failing cases with `assertFailsWith` manually. Tests on the old JVM backend will be removed soon anyway.